app/models/quiz.rb

```diff
@@ -3,4 +3,19 @@ class Quiz < ActiveRecord::Base
 
   validates :name, presence: true
   validates :name, uniqueness: true
+
+  def completed_by_student?(user)
+    questions.all? do |q|
+      completed_answers = AnswerSubmission.
+        includes(:question).
+        where(user: user, questions: { quiz_id: self.id })
+
+      !completed_answers.empty?
+    end
+  end
```

Comment:
I would be a good idea to have a few model tests for this method. One for a user who has not answered any questions, one for a user who has answered some of the questions, and one for a user who has answered all the questions.